api/v1alpha1/nginxingresscontroller_types.go

```diff
@@ -146,7 +153,8 @@ type NginxIngressController struct {
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	// +required
-	Spec NginxIngressControllerSpec `json:"spec,omitempty"`
+	// +kubebuilder:default:={"ingressClassName":"nginx.approuting.kubernetes.azure.com","controllerNamePrefix":"nginx"}
```

Comment:
If this string is malformed our e2e runner will fail because the crd won't be able to be applied. 
CRDs validate this when they are applied against the k8s api server.

I know because this happened to me when I made a typo in an earlier iteration of this PR.